src/Microsoft.AspNet.OData.Shared/Extensions/ContainerBuilderExtensions.cs

```diff
@@ -48,8 +48,11 @@ public static IContainerBuilder AddDefaultWebApiServices(this IContainerBuilder
             builder.AddService<OrderByQueryValidator>(ServiceLifetime.Singleton);
             builder.AddService<SelectExpandQueryValidator>(ServiceLifetime.Singleton);
             builder.AddService<SkipQueryValidator>(ServiceLifetime.Singleton);
+            builder.AddService<SkipTokenQueryValidator>(ServiceLifetime.Singleton);
             builder.AddService<TopQueryValidator>(ServiceLifetime.Singleton);
 
+            builder.AddService<SkipTokenHandler, DefaultSkipTokenHandler>(ServiceLifetime.Singleton);
```

Comment:
With the current implementation, the SkipTokenHandler will hold a state for the current request. If the scope is singleton, concurrent requests will interfere with each other and will lead to issues.